packages/excavator-script/src/projects/designerSweatpants.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * @author horrible-little-slime
+ * Track sweatiness of various sweatpants zones
+ */
+import { currentRound, equippedItem, myLocation } from "kolmafia";
+import { $item, $slot } from "libram";
+
+import { ExcavatorProject } from "../type";
+
+type SweatpantsData = { location: number; sweat: number };
+const SWEAT_PATTERN = /You get (\d)% sweatier/;
+function spadeSweatpants(
+  _encounter: string,
+  page: string,
+): SweatpantsData | null {
+  if (currentRound() !== 0) return null;
+  if (equippedItem($slot`pants`) !== $item`designer sweatpants`) return null;
+  const result = page.match(SWEAT_PATTERN);
+  if (result) return { location: myLocation().id, sweat: Number(result[1]) };
+  return null;
+}
```

Comment:
This suggestion is a combination of my preferred structure and matching the shape of other projects.

```suggestion
function spadeSweatpants(
  encounter: string,
  page: string,
) {
  if (currentRound() !== 0) return null;
  if (equippedItem($slot`pants`) !== $item`designer sweatpants`) return null;
  const sweat = page.match(/You get (\d)% sweatier/)?.[1];
  if (!sweat) return null;
  return { location: myLocation().id, sweat: Number(sweat) };
}
```